packages/schematics/angular/migrations/update-6/index.ts

```diff
@@ -0,0 +1,384 @@
+      config.schematics = schematicsConfig;
+    }
+    const architectConfig = extractArchitectConfig(oldConfig);
+    if (architectConfig !== null) {
+      config.architect = architectConfig;
+    }
+
+    context.logger.info(`Removing old config file (${oldConfigPath})`);
+    host.delete(oldConfigPath);
+    context.logger.info(`Writing config file (${configPath})`);
+    host.create(configPath, JSON.stringify(config, null, 2));
+
+    return host;
+  };
+}
+
+function extractCliConfig(_config: CliConfig): JsonObject | null {
+  return null;
+}
+
+function extractSchematicsConfig(config: CliConfig): JsonObject | null {
+  let collectionName = '@schematics/angular';
+  if (!config || !config.defaults) {
+    return null;
+  }
+  // const configDefaults = config.defaults;
+  if (config.defaults && config.defaults.schematics && config.defaults.schematics.collection) {
+    collectionName = config.defaults.schematics.collection;
+  }
+
+  /**
+   * For each schematic
+   *  - get the config
+   *  - filter one's without config
+   *  - combine them into an object
+   */
+  // tslint:disable-next-line:no-any
+  const schematicConfigs: any = ['class', 'component', 'directive', 'guard',
+                                 'interface', 'module', 'pipe', 'service']
+    .map(schematicName => {
+      // tslint:disable-next-line:no-any
+      const schematicDefaults: JsonObject = (config.defaults as any)[schematicName] || null;
+
+      return {
+        schematicName,
+        config: schematicDefaults,
+      };
+    })
+    .filter(schematic => schematic.config !== null)
+    .reduce((all: JsonObject, schematic) => {
+      all[collectionName + ':' + schematic.schematicName] = schematic.config;
+
+      return all;
+    }, {});
+
+  const componentUpdate: JsonObject = {};
+  componentUpdate.prefix = '';
+
+  const componentKey = collectionName + ':component';
+  const directiveKey = collectionName + ':directive';
+  if (!schematicConfigs[componentKey]) {
+    schematicConfigs[componentKey] = {};
+  }
+  if (!schematicConfigs[directiveKey]) {
+    schematicConfigs[directiveKey] = {};
+  }
+  if (config.apps && config.apps[0]) {
+    schematicConfigs[componentKey].prefix = config.apps[0].prefix;
+    schematicConfigs[directiveKey].prefix = config.apps[0].prefix;
+  }
+  if (config.defaults) {
+    schematicConfigs[componentKey].styleext = config.defaults.styleExt;
+  }
+
+  return schematicConfigs;
+}
+
+function extractArchitectConfig(_config: CliConfig): JsonObject | null {
+  return null;
+}
+
+function extractProjectsConfig(config: CliConfig): JsonObject {
+  const builderPackage = '@angular-devkit/build-webpack';
+
+  const apps = config.apps || [];
+  // convert the apps to projects
+  const projectMap = apps
+    .map((app: AppConfig, idx: number) => {
+      const name = app.name || `app${idx + 1}`;
+      const outDir = app.outDir || defaults.outDir;
+      const projectRoot = app.root || defaults.appRoot;
```

Comment:
`root` in old projects is `src/`, while in new projects it shouldn't be because we imply `src/` already in other parts.

Basically `ng new x` would set the project root to `projects/x/` with src being `projects/x/src/`, but using the `app.root` here makes it `src/` with src being `src/src/`. So you cannot generate any component or schematics in general if you keep this.